services/pull/review.go

```diff
@@ -283,6 +348,15 @@ func createCodeComment(ctx context.Context, doer *user_model.User, repo *repo_mo
 			log.Error("Error whilst generating patch: %v", err)
 			return nil, err
 		}
+
+		// If patch is still empty (unchanged line), generate code context
+		if len(patch) == 0 && len(commitID) > 0 {
+			patch, err = generatePatchForUnchangedLine(gitRepo, commitID, treePath, line, setting.UI.CodeCommentLines)
+			if err != nil {
+				log.Warn("Error generating patch for unchanged line: %v", err)
+				// Don't fail comment creation, just leave patch empty
```

Comment:
Maybe we could return an error to user so that we don't need to check the patch in `CreateComment` again?